WeatherForecast/WeatherForecast/WeatherDetail/WeatherDetailView.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  WeatherDetailView.swift
+//  WeatherForecast
+//
+//  Created by Daegeon Choi on 2024/02/03.
+//
+
+import Foundation
+import UIKit
+
+protocol WeatherDetailDelegate {
+    var navigationItem: UINavigationItem { get }
+}
+
+class WeatherDeatilView: UIView {
+    private var delegate: WeatherDetailDelegate
```

Comment:
delegate를 채택하고 있는 클래스가 WeatherDetailViewController 이기 때문에 서로 강한 참조를 가지게 될 것 같아요

일반적으로는 아래와 같이 사용합니다.
```
protocol WeatherDetailDelegate: **AnyObject**
```

```suggestion
    private weak var delegate: WeatherDetailDelegate?
```